CONTRIBUTING.md

```diff
@@ -0,0 +1,72 @@
+# Contribution Guidelines
+
+Thank you for your interest in contributing to our project :)<br>
+We like issue reports, suggestions and pull request from everyone.
+
+If you are planning to make a major contribution for which there is no existing issue, we would appreciate that you open an issue first.
+
+The following suggestions increase the chance that your pull request can be accepted:
+
+-   Write clean code
+-   Write tests for your new code
+-   Check that all tests are still passed
+-   Adhere to the naming conventions for branch names and commit messages mentioned below
+-   Avoid unnecessary breaking changes
+
+You and your code contribution has to adherence to the licence of this project.
+
+## Naming Conventions
+
+### Branch Names
+
+Branch names should follow the following structure: `<type>/<issue-id>/<name>`
+
+-   `<type>` is the type of the change (see table below).
+-   `<number>` is the number of the corresponding issue on GitHub the branch should resolve.
+-   `<name>` is a describing branch name, which should be in **lowercase** separated by **dashes**.
+
+### Allowed Change Types
+
+| Change Type | Description                            |
+| ----------- | -------------------------------------- |
+| `feat`      | Added new or expanded existing feature |
+| `fix`       | Bugfixes                               |
+| `docs`      | Updated documentation only             |
+| `revert`    | Revert of some commit                  |
+| `style`     | Code style improvements                |
+| `refactor`  | Major refactorings                     |
+| `tech`      | Other kinds of technical changes       |
+
+Examples:
+
+-   `feat/123/settings-option-xyz`
+-   `fix/124/unexpected-settings-error`
+
+### Commit Messages
+
+Commit messages should follow the [Conventional Commits standard](https://www.conventionalcommits.org/en/v1.0.0/) with the following additional rules:
+
+-   The first line MUST end with the id of the corresponding issue on GitHub
+-   The type MUST be one of the types in the table "allowed change types" above
```

Comment:
Conventional Commits does allow more than the mentioned change types. e.g. chore.
Is there a reason to limit the change types?

Reply:
I would try to stick to the standard as long as there is a good reason to deviate from it.

Reply:
Well, there is no standard change type other than "feat" and "fix". I would at least like to include the types similar to the ones used by the branch names in CodeCharta.

Reply:
I have now changed the "change types" to exactly match the recommendation found under the link I mentioned above, as I guess that you meant that.